### PR TITLE
[docs] Global pause for regulated coin update

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

Adds global pause documentation and updates to v2 naming.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: docs/content/standards/coin.mdx [hunk 1]
```diff
@@ -217,33 +217,18 @@ Use the following functions to get the values for the respective fields on the m
...
-
-```move
-public fun deny_list_contains<T>(
-  freezer: &deny_list::DenyList,
-  addr: address
-): bool
-```
-
-The function returns `true` if the address is found on the coin's list, otherwise it returns `false`.
-
```
Comment (cgswords) [inline docs/content/standards/coin.mdx:232]: Did we remove this feature, or document `deny_list_v2_contains_current_epoch` and `deny_list_v2_contains_next_epoch`?

**Review by cgswords — APPROVED** [review]
LGTM but tagging @tnowacki  since he wrote most of the code. One small question outstanding.

File: docs/content/standards/coin.mdx (hunk 1 above)
Comment (ronny-mysten) [reply docs/content/standards/coin.mdx:232]: It's mostly for internal (per TN) use so removed.

File: docs/content/standards/coin.mdx [hunk 2]
```diff
@@ -67,72 +75,47 @@ The fields of the metadata objects include the following:
...
 
 The signature shows that only the `TreasuryCap` and coin object you want to burn are necessary inputs, returning the amount by which the supply was decreased (value of the coin). The function does not allow you to burn more coins than are available in the supply.
 
 ## Adding and removing addresses to and from the deny list
 
-The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCap` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions.
+The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCapV2` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions starting immediately upon being added. At the following epoch after addresses get added to the list, those addresses cannot receive the coin type.
```
Comment (tnowacki) [inline docs/content/standards/coin.mdx:102]:
```suggestion
The deny list is only applicable to regulated coins. As mentioned previously, when you create a regulated coin you receive a `DenyCapV2` that authorizes the bearer to add and remove addresses from the system-created `DenyList` object. Any address on the list for your coin is unable to use the coin as an input to transactions starting immediately upon being added. At the following epoch after addresses get added to the list, those addresses additionally cannot receive the coin type.
```
Thoughts on making it clearer that this is additional behavior? i.e. that it is not that it somehow switches from sending to receiving.

File: docs/content/standards/coin.mdx [hunk 3]
```diff
@@ -40,7 +40,15 @@ The [regulated-coin-sample repository](https://github.com/MystenLabs/regulated-c
...
 
-The list of addresses that aren't able to use a particular regulated coin is held within a system-created `DenyList` shared object. If you have access to the `DenyCap`, then you can use the `coin::deny_list_add` and `coin::deny_list_remove` functions to add and remove addresses. You can also use the `coin::deny_list_contains` function to check if an address is already on the list.
+The list of addresses that aren't able to use a particular regulated coin is held within a system-created `DenyList` shared object. If you have access to the `DenyCap`, then you can use the `coin::deny_list_v2_add` and `coin::deny_list_v2_remove` functions to add and remove addresses.
+
+### Global pause switch
+
+Regulated coin objects include an `allow_global_pause` Boolean field. When set to `true`, the bearer of the `DenyCapV2` object for the coin type can use the `coin::deny_list_v2_enable_global_pause` function to pause coin activity indefinitely. Immediately upon the bearer initiating the pause, the network disallows the coin type as input for any transactions. At the start of the next epoch (epochs last ~24 hours), the network disallows all addresses from receiving the coin type.
```
Comment (tnowacki) [inline docs/content/standards/coin.mdx:47]:
```suggestion
Regulated coin objects include an `allow_global_pause` Boolean field. When set to `true`, the bearer of the `DenyCapV2` object for the coin type can use the `coin::deny_list_v2_enable_global_pause` function to pause coin activity indefinitely. Immediately upon the bearer initiating the pause, the network disallows the coin type as input for any transactions. At the start of the next epoch (epochs last ~24 hours), the network additionally disallows all addresses from receiving the coin type. 
```
Similar to the other change

File: docs/content/standards/coin.mdx (hunk 2 above)
Comment (ronny-mysten) [reply docs/content/standards/coin.mdx]: Good call. I also like the general "sending" and "receiving" language, so I additionally added it.

File: docs/content/standards/coin.mdx (hunk 3 above)
Comment (ronny-mysten) [reply docs/content/standards/coin.mdx]: Dig it.

**Pull request merged**